noir_stdlib/src/uint128.nr

```diff
@@ -91,23 +99,36 @@ impl U128 {
         if ascii < 58 {
             ascii - 48
         } else if ascii < 71 {
+            assert(ascii >= 65); // enforce >= 'A'
```

Comment:
If you want to validate the input, I'd rather do something like this, which avoids two range checks:
```
    unconstrained fn upper(ascii: u8) -> bool {
        (ascii < 71)
    }
    fn decode_ascii(ascii: u8) -> Field {
         if ascii < 58 {
            ascii - 48
        } else {
	        let ascii = ascii + 32 * (U128::upper(ascii) as u8);
            assert(ascii >= 97); // enforce >= 'a'
            assert(ascii <= 102); // enforce <= 'f'
            ascii - 87
        } as Field
    }
```